### PR TITLE
Csl865/crypto tests

### PR DESCRIPTION
**Pull request by rockbmb**

### CODE REVIEW

File: test/Test/Pos/CryptoSpec.hs [hunk 1]
```diff
@@ -260,3 +268,78 @@ redeemThenCheckDifferentData sk a b =
...
+    -> ByteString
+    -> Property
+encrypyDecryptChaChaDifferentKey
+    (Nonce nonce)
+    (Crypto.HDPassphrase key1)
+    (Crypto.HDPassphrase key2)
+    header
+    plaintext =
+    (key1 /= key2) ==>
+    (isLeft (join  (decrypt <$> (Crypto.toEither . encrypt $ plaintext))))
```
Comment (ChShersh) [inline test/Test/Pos/CryptoSpec.hs]: `join (f <$> m)` is `m >>= f`

File: test/Test/Pos/CryptoSpec.hs [hunk 2]
```diff
@@ -362,3 +411,122 @@ encToPublicToEnc =
 skToSafeSigner :: Crypto.SecretKey -> Property
 skToSafeSigner =
     Crypto.safeToPublic . Crypto.fakeSigner .=. Crypto.toPublic
+
+verifyEncShareGoodData :: SharedSecrets -> Bool
+verifyEncShareGoodData SharedSecrets
+    { getSecretSharing = secShare
```
Comment (ChShersh) [inline test/Test/Pos/CryptoSpec.hs]: This formatting and pattern matching by records doesn't look pretty... Why not use `RecordsWildCards` here instead? Also, seems like `SharedSecrets` fields are not named by the convention in out style-guide.

**Comment by rockbmb** [comment]
@ChShersh Can you take a look now? Should be finished this time.

**Comment by rockbmb** [comment]
@ChShersh don't merge it just yet, I need to figure out why CI failed.

**Comment by ChShersh** [comment]
@rockbmb I've approved. Seems ok. But I see that build fails. Though I don't know whether its `master` error. But currently I can't merge PR due to failed checks.

**Comment by rockbmb** [comment]
@ChShersh Fixed, it was a problem with `<<<<<<<` I forgot to remove from a merge. CI should work now, then you can merge 👍

**Comment by ChShersh** [comment]
Still some errors in build :(

**Comment by rockbmb** [comment]
@ChShersh Yeah, we're all here at the office and it appears CI is erroring when it shouldn't. It did indeed pass for Travis. We'll have to find a workaround, but for now I guess we're stuck.

I can assure you, however, that the tests work fine now.

**Comment by rockbmb** [comment]
@ChShersh After CI passes (I've merged Domen's latest fix), please merge. At least I hope it passes.

**Comment by gromakovsky** [comment]
@ChShersh CI passes btw.

**Comment by rockbmb** [comment]
@ChShersh yoooo

**Comment by gromakovsky** [comment]
Hello, @ChShersh, how are you? Just wanted to ask you to pay attention to this pull request.

**Pull request merged**